core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java

```diff
@@ -219,16 +221,24 @@ public void start() {
                availabilityTask = nonBlockingManager.scheduleWithFixedDelay(this::pollStoreAvailability, interval, interval, MILLISECONDS));
          }
 
-         storeStartup.doOnComplete(() -> lock.unlockWrite(stamp))
-               // Blocks here waiting for stores and availability task to start if needed
-               .blockingAwait();
+         // Blocks here waiting for stores and availability task to start if needed
+         storeStartup.blockingAwait();
+         allSegmentedOrShared = allStoresSegmentedOrShared();
       } catch (Throwable t) {
          lock.unlockWrite(stamp);
          log.debug("PersistenceManagerImpl encountered an exception during startup of stores", t);
          throw t;
+      } finally {
+         lock.unlockWrite(stamp);
       }
    }
 
+   @GuardedBy("lock")
+   private boolean allStoresSegmentedOrShared() {
+      return getStoreLocked(storeStatus -> !storeStatus.characteristics.contains(Characteristic.SEGMENTABLE) ||
+            !storeStatus.characteristics.contains(Characteristic.SHAREABLE)) != null;
```

Comment:
This seems wrong, just because a store is segmentable and shareable it doesn't mean in the config it's actually segmented or shared (as the method name implies).

Reply:
At this point the store status has already been changed by `updateCharacteristics` which will remove the characteristic based on the configuration. So it does mean that.

Reply:
I would have never guessed that... wouldn't it have been simpler to check the configuration instead of changing the meaning of the enum values (e.g. SHAREABLE -> SHARED)?

Reply:
Yes, I agree. I noticed this a while back as I had been changing things, but I wanted to try to limit changes for now. I am all for changing this later though.

Reply:
:+1: for later :)